lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/common/VortexFutureDelegate.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.vortex.common;
+
+import org.apache.reef.annotations.Unstable;
+import org.apache.reef.annotations.audience.DriverSide;
+import org.apache.reef.annotations.audience.Private;
+
+import java.io.Serializable;
+import java.util.List;
+
+/**
+ * Exposes functions to be called by the {@link org.apache.reef.vortex.driver.VortexMaster}
+ * to note that a list of Tasklets associated with a Future has completed.
+ */
+@Unstable
+@DriverSide
+@Private
+public interface VortexFutureDelegate<TOutput extends Serializable> {
```

Comment:
There is currently no examples to specify the local aggregator function yet, but when we submit a set of VortexFunctions and their inputs, we can optionally associate the set of VortexFunctions with an AggregatorFunction, which will aggregate completed Tasklets according to a aggregation policy.
As an example, suppose we have 10 Tasklets associated with the same AggregationFunction. If 5 Tasklets succeeded and 2 Tasklets failed when the AggregationPolicy triggers, the aggregation function will aggregate on the 5 Tasklets that have succeeded. In this `WorkerReport`, the Worker will generate 3 `TaskletReports` -- the succeeded aggregation of 5 Tasklets, and 2 failed Tasklet reports. Each of the failed Tasklets will have its own Exception associated with it. Suppose on the next round of aggregation policy trigger, the remaining 3 Tasklets report back as succeeded, and we run the AggregationFunction on the 3 Tasklet results. If the AggregationFunction throws an Exception this time because an output of one of the Tasklets were unexpected, the `WorkerReport` will only contain a single `TaskletReport`, with the List of TaskletIds containing the 3 Tasklets that failed on aggregation, and the Exception that has caused the aggregation to fail.